cilium-cli/connectivity/sniff/sniffer.go

```diff
@@ -25,17 +25,105 @@ const (
 	// ModeSanity: expect to observe packets matching the filter, to be
 	// leveraged as a sanity check to verify that the filter is correct.
 	ModeSanity Mode = "sanity"
+
+	// Max wait time for tcpdump to start/stop in remote shell.
+	sniffScriptTimeout = 10 * time.Second
+	// Max timeout before aborting k8s connection in Start/Stop; must exceed sniffScriptTimeout.
+	sniffConnectionTimeout = sniffScriptTimeout + 5*time.Second
+	// Max remote sniffer runtime to prevent lingering processes in the pod.
+	// NOTE: too low may kill tcpdump while test is running.
+	sniffKillTimeout = sniffConnectionTimeout * 4
```

Comment:
This is problematic, as there are tests that may require a long-running tcpdump capture, e.g., to verify that certain packets are not seen during the entire connectivity suite execution. I general I don't think it is possible to reliably define a maximum duration of the capture.

Reply:
> One that comes to my mind is the one that looks for unencrypted leaks, but it's based on bpftrace, not tcpdump.

The current timeout is at 1 min. Won't that be way too low for the tests you mention?